discovery/src/test/java/org/ethereum/beacon/discovery/DiscoveryInteropTest.java

```diff
@@ -0,0 +1,121 @@
+package org.ethereum.beacon.discovery;
+
+import org.ethereum.beacon.db.Database;
+import org.ethereum.beacon.discovery.enr.NodeRecord;
+import org.ethereum.beacon.discovery.packet.AuthHeaderMessagePacket;
+import org.ethereum.beacon.discovery.packet.RandomPacket;
+import org.ethereum.beacon.discovery.packet.UnknownPacket;
+import org.ethereum.beacon.discovery.storage.NodeBucket;
+import org.ethereum.beacon.discovery.storage.NodeBucketStorage;
+import org.ethereum.beacon.discovery.storage.NodeTableStorage;
+import org.ethereum.beacon.discovery.storage.NodeTableStorageFactoryImpl;
+import org.ethereum.beacon.schedulers.Schedulers;
+import org.javatuples.Pair;
+import org.junit.Ignore;
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.ethereum.beacon.discovery.TestUtil.NODE_RECORD_FACTORY_NO_VERIFICATION;
+import static org.ethereum.beacon.discovery.TestUtil.TEST_SERIALIZER;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Inter-operational test with Geth. Start it in docker separately
+ *
+ * <p>You need to build and run Geth discv5 test to interact with. Configure Geth running time in
+ * test.sh located in `resources/geth`, after that build docker and run it: <code>
+ *   cd discovery/src/test/resources/geth
+ *   docker build -t gethv5:1.0 . && docker run --network host -d gethv5:1.0
+ * </code>
+ *
+ * <p>After container starts, fire this test to fall in Geth's side running time and it should pass!
+ * You could check Geth test logs by following command:<code>
+ *   docker logs <container-id/>
+ * </code>
+ */
+@Ignore("Requires manual startup, takes a bit to start")
+public class DiscoveryInteropTest {
+  @Test
+  public void testInterop() throws Exception {
+    // 1) start 2 nodes
+    Pair<BytesValue, NodeRecord> nodePair1 = TestUtil.generateNode(40412, true);
+    System.out.println(String.format("Node %s started", nodePair1.getValue1().getNodeId()));
+    NodeRecord nodeRecord1 = nodePair1.getValue1();
+    NodeRecord nodeRecord2 =
+        NODE_RECORD_FACTORY_NO_VERIFICATION.fromBase64(
```

Comment:
It would be convenient if one may change parameters of this node like IP address. I had to replace this base64 line to make test run on Mac OS as it handles docker in a `docker-machine` instance with its own network interface and addresses.

Reply:
Yeah, did this, plus refactored a bit NodeRecord creation